docs/excel/custom-functions-naming.md

```diff
@@ -0,0 +1,38 @@
+---
+ms.date: 02/08/2019
+description: Learn requirements for Excel custom functions' names and avoid common naming pitfalls.
+title: Naming guidelines for custom functions in Excel (preview)
+localization_priority: Normal
+---
+# Naming guidelines
+
+Custom functions contain two unique properties in their JSON metadata file: their function name and their function ID. Function IDs are used as the main way to differentiate functions from each other in Excel programmatically. Function names are what appear to users of your add-in as the function's name in Excel. Function names can differ from the ID, e.g. for localization purposes, but should generally stay the same as the ID if there is not a compelling reason for them to differ.
+
+Function names and function IDs share some common requirements:
+
+- Can use alphanumeric characters (including Unicode), the numbers zero through nine, underscores, and periods.
+
+- Must start with a letter and have a minimum limit of three characters.
+
+Most function names for built-in Excel functions appear in uppercase letters (e.g. `SUM`). While not required, it is a best practice to specify your custom function names and function IDs using uppercase letters.
+
+Function names should not be named the same as:
+
+- Any cells between A1 to XFD1048576 or any cells between R1C1 to R1048576C16384.
+
+- Any Excel 4.0 Macro Function (such as `RUN`, `ECHO`).  For a full list of these functions, see [this article](https://www.microsoft.com/en-us/download/details.aspx?id=1465).
+
+## Naming conflicts
+
+If a function name in your add-in matches a name in an existing and installed add-in, you'll see an error #REF! appear in your workbook.
+
+To troubleshoot this, please change the name in your add-in and try the function again. If you're testing your add-in in different environments, either uninstall the add-in with the conflicting name or use a different namespace to differentiate your function (e.g. NAMESPACE_NAMEOFFUNCTION).
+
+Also consider how you'd like people to use the functions within your add-in. In many cases, it makes sense to add multiple arguments to a function rather than create multiple functions with the same or similar names.
```

Comment:
Would be good to show an example here of "instead of this... do this..."

Reply:
I agree with David, and example or two would be great here. When do you add arguments and how do you name them. When do you NOT add arguments. For an example, I once wrote a custom function that returned the size of artworks in our database: GETARTDIMENSIONS(height, width, length) instead of GETARTHEIGHT, GETARTLENGHT, GETARTWIDTH. 

I also think you should consider adding a few more "best practices" sentences about naming functions and arguments so they describe what they do and the importance of readability. They should be named using natural language so others can understand.
1. They shouldn't be pointless like X1, FE2or XBQNE or overly long either, like ADDNEWIFLEGALDRINKINGAGEANDMOONISFULL. They should tell it like it is i.e. ISLEGALDRINKINGAGE. 
2. Avoid ambiguous abbreviations. Clarity is more important than shortness. i.e. choose INCREMENT(), INCREASE() or INCLUDE(), not just INC(). 
3. If a function is an action, it should include the verb in its name: i.e. GETZIPCODE or SETZIPCODE instead of just ZIPCODE. 
4. You might also add a link to some naming schemes such as how to indicate what the function returns. Prefix with S_ for a string function, B_ for boolean, that sort of thing. (Though allcaps instead of lower camel case makes it hard to use Hungarian!) 
6. Shouldn't the function names should be in English--since programming languages are in English, it makes sense, no?)
7. Choose one name for an action and stick with it, don't use both REMOVEZIP and DELETEADDRESS, select one and stick with it. i.e.  REMOVEZIP and REMOVEADDRESS.